src/optimizer/memo.cpp

```diff
@@ -43,8 +43,6 @@ GroupExpression *Memo::InsertExpression(std::shared_ptr<GroupExpression> gexpr,
   auto it = group_expressions_.find(gexpr.get());
 
   if (it != group_expressions_.end()) {
-    PL_ASSERT(target_group == UNDEFINED_GROUP ||
-           target_group == (*it)->GetGroupID());
```

Comment:
I debugged for a while and reached the conclusion that this assert was incorrect. From my understanding, it doesn't make sense that if we create a group expression, we search for it in the memo table, and we find an existing equal expression, that we should have the same `GroupID`. Consider two possible join trees (AxB)x(CxD) and Ax(Bx(CxD)). In these two cases, both of the cases will generate a group expression for (CxD) independently.